### PR TITLE
fix: updated readme

### PR DESCRIPTION
**Pull request by Ankush1oo8**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)
fixed issue #485 

Update readme to reflect support for fastify 5. Support for v5 came with 8.0.0 release

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,8 @@ Plugin for serving static files as fast as possible. Supports Fastify version `4
 
 | Fastify version | branch                                                               |
 | --------------- | -------------------------------------------------------------------- |
-| `^4.x`          | This branch                                                          |
+| `^5.x`          | This branch
```
Comment (dancastillo) [inline README.md:11]: I see it did say `This branch` IMHO I think we should follow other repos like autoload [here](https://github.com/fastify/fastify-autoload?tab=readme-ov-file#compatibility)

```suggestion
| `^5.x`          | [`v8.x`](https://github.com/fastify/fastify-static/tree/master)
```

Comment (Ankush1oo8) [reply README.md:11]: Should I click on resolve or manually commit the change

Comment (dancastillo) [reply README.md:11]: I think either or is fine

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**